### PR TITLE
PM-9439: Update cipher list item for passkeys

### PR DESCRIPTION
**Pull request by shannon-livefront**
## 🎟️ Tracking

[PM-9439](https://bitwarden.atlassian.net/browse/PM-9439)

## 📔 Objective

Update the default icon of passkey items during the passkey creation and autofill flows, and, if the RpId of the passkey doesn't match the passkey's name, add the RpId to the subtitle of the item.

## 📸 Screenshots

| Before | After |
| --- | --- |
|  |  |

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-9439]: https://bitwarden.atlassian.net/browse/PM-9439?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt [hunk 1]
```diff
@@ -523,6 +524,20 @@ class VaultRepositoryImpl(
...
+    override suspend fun getDecryptedFido2CredentialAutofillViews(
+        cipherViewList: List<CipherView>,
+    ): List<Fido2CredentialAutofillView> {
+        return vaultSdkSource
+            .decryptFido2CredentialAutofillViews(
+                userId = activeUserId ?: return emptyList(),
+                cipherViews = cipherViewList.toTypedArray(),
+            )
+            .fold(
+                onFailure = { emptyList() },
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt:536]: So we do not want to report a failure here and always return a list?

Usually we would create a sealed class for this where only the success has the list of data.

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/util/VaultItemListingDataExtensions.kt [hunk 2]
```diff
@@ -287,32 +297,48 @@ private fun List<SendView>.toDisplayItemList(
...
     )
 
+private fun CipherView.toSubtitle(fido2CredentialRpId: String?): String? =
+    fido2CredentialRpId
+        ?.takeIf { this.type == CipherType.LOGIN && it.isNotEmpty() && it != this.name }
+        ?.let {
+            if (subtitle.isNullOrEmpty()) {
+                it
+            } else {
+                it + "\n" + subtitle
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/util/VaultItemListingDataExtensions.kt:337]: Would it make more sense to just add a new nullable property and display it in the composable when present. Then we can avoid the string concatenation stuff here.

File: app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt [hunk 3]
```diff
@@ -861,22 +876,6 @@ class VaultRepositoryImpl(
             )
     }
 
-    /**
-     * Return a filtered list containing elements that match the given [relyingPartyId] and a
-     * credential ID contained in [credentialIds].
-     */
-    private fun List<Fido2CredentialAutofillView>.filterMatchingCredentials(
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt:868]: 👍

File: app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/item/util/CipherViewExtensionsTest.kt [hunk 4]
```diff
@@ -177,7 +178,7 @@ class CipherViewExtensionsTest {
                     .copy(currentCipher = cipherView),
                 type = createLoginContent(isEmpty = false).copy(
                     isPremiumUser = isPremiumUser,
-                    canViewTotpCode = true
+                    canViewTotpCode = true,
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/item/util/CipherViewExtensionsTest.kt:181]: How is this stuff making it past CI

Thanks for fixing it!

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/util/VaultItemListingDataExtensions.kt (hunk 2 above)
Comment (shannon-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/util/VaultItemListingDataExtensions.kt:337]: The Figma designs showed the subtitle as a single multi-line text box so I thought that was what we wanted. I can ask about it!

File: app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt (hunk 1 above)
Comment (shannon-livefront) [reply app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt:536]: Ok, I can do that!

File: app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt [hunk 5]
```diff
@@ -523,6 +524,20 @@ class VaultRepositoryImpl(
...
+        cipherViewList: List<CipherView>,
+    ): DecryptFido2CredentialAutofillViewResult {
+        return vaultSdkSource
+            .decryptFido2CredentialAutofillViews(
+                userId = activeUserId ?: return DecryptFido2CredentialAutofillViewResult.Error,
+                cipherViews = cipherViewList.toTypedArray(),
+            )
+            .fold(
+                onFailure = { DecryptFido2CredentialAutofillViewResult.Error },
+                onSuccess = { DecryptFido2CredentialAutofillViewResult.Success(it) },
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/vault/repository/VaultRepositoryImpl.kt:537]: 👍

File: app/src/main/java/com/x8bit/bitwarden/data/vault/repository/model/DecryptFido2CredentialAutofillViewResult.kt [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Models result of decrypting the fido2 credential autofill views.
+ */
+sealed class DecryptFido2CredentialAutofillViewResult {
+    /**
+     * Credentials decrypted successfully.
+     */
+    data class Success(
+        val fido2CredentialAutofillViews: List<Fido2CredentialAutofillView>,
+        ) : DecryptFido2CredentialAutofillViewResult()
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/vault/repository/model/DecryptFido2CredentialAutofillViewResult.kt:14]: Wanna format this

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModel.kt [hunk 7]
```diff
@@ -895,6 +899,14 @@ class VaultItemListingViewModel @Inject constructor(
                     ciphers = vaultData.cipherViewList,
                     matchUri = matchUri,
                 ),
+                fido2CredentialAutofillViewList = (vaultRepository
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModel.kt:902]: 💭 Could we put this filter logic in its own private function and use it here and on `line 930` too.

**Review by dseverns-livefront — DISMISSED** [review]
One non-blocking thought for me, also "Baby's first WebAuthn" 😆

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModel.kt (hunk 7 above)
Comment (shannon-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/VaultItemListingViewModel.kt:902]: I like that - thanks!

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/util/VaultItemListingDataExtensions.kt (hunk 2 above)
Comment (shannon-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/itemlisting/util/VaultItemListingDataExtensions.kt]: Got the update from the team and changed the text field and test tag accordingly! 👌

**Review by david-livefront — APPROVED** [review]

**Review by david-livefront — APPROVED** [review]

**Pull request merged**